store/tikv/coprocessor.go

```diff
@@ -977,6 +995,28 @@ func (worker *copIteratorWorker) handleCopResponse(bo *Backoffer, rpcCtx *RPCCon
 			}
 		}
 	}
+	if resp.pbResp.IsCacheHit && cacheValue != nil {
+		// Cache hit and is valid: use cached data as response data and we don't update the cache.
+		data := make([]byte, len(cacheValue.Data))
+		copy(data, cacheValue.Data)
+		resp.pbResp.Data = data
+	} else {
+		// Cache not hit or cache hit but not valid: update the cache if the response can be cached.
+		if cacheKey != nil && resp.pbResp.CacheLastVersion > 0 {
```

Comment:
From TiKV perspective, yes, since TiKV only compares the CacheVersion and does not know which region ID TiDB is cached.

From TiDB perspective, no, since the cache is only supplied when RequestRegionID matches the RegionID in the cache:

```
if cacheValue != nil && cacheValue.RegionID == task.region.id && cacheValue.TimeStamp <= worker.req.StartTs
```